### PR TITLE
[17515] Use standard value for PID_RELATED_SAMPLE_IDENTITY

### PR DESCRIPTION
**Pull request by MRicoIE2CS**
## Description

The value for PID_RELATED_SAMPLE_IDENTITY enum definition is changed to (`0x0083`) the one defined in the document `Remote Procedure Call over DDS`, document reference `ptc/2016-03-19`, V1.0.
The legacy value used is now defined under PID_CUSTOM_RELATED_SAMPLE_IDENTITY.
The definitions for the values `0x0080` to `0x0082` have been added as well.

## Contributor Checklist
- [X] Commit messages follow the project guidelines. 
- [X] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [X] Any new/modified methods have been properly documented using Doxygen. 
- [X] Changes are ABI compatible. 
- [X] Changes are API compatible. 
- [N/A] New feature has been added to the `versions.md` file (if applicable).
- [N/A] New feature has been documented/Current behavior is correctly described in the documentation. 
    
- [N/A] Applicable backports have been included in the description.

## Reviewer Checklist
- [ ] The PR has a milestone assigned.
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by MRicoIE2CS** [comment]
@richiprosima please test

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]
We also need to send this new property ID along with the old one.

**Comment by MRicoIE2CS** [comment]
@richiprosima please test

**Comment by MRicoIE2CS** [comment]
@richiprosima please test

**Comment by jsantiago-eProsima** [comment]
I think we should account for the added paremeter [in this size calculation](https://github.com/eProsima/Fast-DDS/blob/7fb237de47653bc2892dae68b29a111e2f7d111c/src/cpp/rtps/history/WriterHistory.cpp#L310)

**Comment by MRicoIE2CS** [comment]
@richiprosima please test this

File: test/blackbox/common/DDSBlackboxTestsBasic.cpp [hunk 1]
```diff
@@ -33,10 +33,18 @@
...
 #include <fastdds/dds/topic/TypeSupport.hpp>
+#include <fastrtps/transport/test_UDPv4TransportDescriptor.h>
 #include <fastrtps/types/TypesBase.h>
+#include <fastdds/core/policy/ParameterSerializer.hpp>
 
 #include "BlackboxTests.hpp"
+#include "../api/dds-pim/PubSubWriter.hpp"
+#include "../api/dds-pim/PubSubReader.hpp"
+//#include "PubSubReader.hpp"
+//#include "PubSubWriter.hpp"
```
Comment (jsantiago-eProsima) [inline test/blackbox/common/DDSBlackboxTestsBasic.cpp:44]: Remove commented lines

File: test/blackbox/common/DDSBlackboxTestsBasic.cpp [hunk 2]
```diff
@@ -271,6 +279,191 @@ TEST(DDSBasic, MultithreadedReaderCreationDoesNotDeadlock)
...
+TEST(DDSBasic, PidRelatedSampleIdentity)
+{
+    PubSubWriter<HelloWorldPubSubType> reliable_writer(TEST_TOPIC_NAME);
+    PubSubReader<HelloWorldPubSubType> reliable_reader(TEST_TOPIC_NAME);
+
+    // Test transport will be used in order to filter inlineQoS
+    auto test_transport = std::make_shared<eprosima::fastrtps::rtps::test_UDPv4TransportDescriptor>();
+    bool exists_pid_related_sample_identity = false;
+    bool exists_pid_custom_related_sample_identity = false;
+    bool another_debug_bool_flag = false;
```
Comment (jsantiago-eProsima) [inline test/blackbox/common/DDSBlackboxTestsBasic.cpp:401]: This does not seem to be used for anything other than passing it to the lambda which then does not use it.

File: test/blackbox/common/DDSBlackboxTestsBasic.cpp [hunk 3]
```diff
@@ -271,6 +279,191 @@ TEST(DDSBasic, MultithreadedReaderCreationDoesNotDeadlock)
...
+
+    test_transport->drop_data_messages_filter_ =
+            [&exists_pid_related_sample_identity, &exists_pid_custom_related_sample_identity,
+                    &another_debug_bool_flag]
+                (eprosima::fastrtps::rtps::CDRMessage_t& msg)-> bool
+            {
+                // Inside this filter, the two flags passed in register wether both PIDs are included in the msg to be sent.
```
Comment (jsantiago-eProsima) [inline test/blackbox/common/DDSBlackboxTestsBasic.cpp:408]:
```suggestion
                // Inside this filter, the two flags passed in register whether both PIDs are included in the msg to be sent.
```

File: test/blackbox/common/DDSBlackboxTestsBasic.cpp [hunk 4]
```diff
@@ -271,6 +279,191 @@ TEST(DDSBasic, MultithreadedReaderCreationDoesNotDeadlock)
...
+                // The legacy value is overwritten in order to sent a msg with only the standard PID_RELATED_SAMPLE_IDENTITY as valid parameter,
```
Comment (jsantiago-eProsima) [inline test/blackbox/common/DDSBlackboxTestsBasic.cpp:409]:
```suggestion
                // The legacy value is overwritten in order to send a msg with only the standard PID_RELATED_SAMPLE_IDENTITY as valid parameter,
```

File: test/blackbox/common/DDSBlackboxTestsBasic.cpp [hunk 5]
```diff
@@ -271,6 +279,191 @@ TEST(DDSBasic, MultithreadedReaderCreationDoesNotDeadlock)
...
+/**
+ * Read a parameterList from a CDRMessage.
+ * Search for PID_CUSTOM_RELATED_SAMPLE_IDENTITY and PID_RELATED_SAMPLE_IDENTITY.
+ * Overwrite PID_CUSTOM_RELATED_SAMPLE_IDENTITY to just leave the new one in msg.
+ * @param[in] msg Reference to the message.
+ * @param[out] exists_pid_related_sample_identity True if the parameter is inside msg.
+ * @param[out] exists_pid_custom_related_sample_identity True if the parameter is inside msg.
+ * @return true if parsing was correct, false otherwise.
+ */
+bool readParameterListfromCDRMsg(
```
Comment (jsantiago-eProsima) [inline test/blackbox/common/DDSBlackboxTestsBasic.cpp:291]: Nitpick: I would change the name of the function to to reflect this just checks for related sample identity.

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

File: test/blackbox/common/DDSBlackboxTestsBasic.cpp [hunk 6]
```diff
@@ -271,6 +275,212 @@ TEST(DDSBasic, MultithreadedReaderCreationDoesNotDeadlock)
...
 }
 
+/**
+* Read a parameterList from a CDRMessage
+* @param[in] msg Reference to the message.
+* @param[out] exists_pid_related_sample_identity True if the parameter is inside msg.
+* @param[out] exists_pid_custom_related_sample_identity True if the parameter is inside msg.
+* @return true if parsing was correct, false otherwise.
+*/
+bool readParameterListfromCDRMsg(
```
[the remark this answers is not in the document]
Comment (MRicoIE2CS) [reply test/blackbox/common/DDSBlackboxTestsBasic.cpp]: You are right with the name, but the doxygen description already stated that. Iguess you did the revision from an early commit.

**Comment by MRicoIE2CS** [comment]
@richiprosima please test this

**Comment by MRicoIE2CS** [comment]
@richiprosima please test this

**Comment by MRicoIE2CS** [comment]
@richiprosima please test this

**Comment by jsantiago-eProsima** [comment]
@richiprosima please test mac

**Review by jsantiago-eProsima — APPROVED** [review]

**Pull request merged**